src/java.net.http/share/classes/jdk/internal/net/http/common/TimeSource.java

```diff
@@ -85,18 +87,19 @@ long delay(long nanos) {
         }
 
         boolean isInWindow(long delay) {
-            return delay >= 0 && delay <= TIME_WINDOW;
+            return delay >= -TIME_WINDOW && delay <= TIME_WINDOW;
         }
 
     }
 
     @Override
     public Deadline instant() {
+        NanoSource source = localSource;
         long nanos = System.nanoTime();
-        long delay = localSource.delay(nanos);
+        long delay = source.delay(nanos);
         // use localSource if possible to avoid a volatile read
```

Comment:
Hello Daniel, the changes look OK to me. However, this comment is now misplaced.